### ISSUE TITLE
[release/2.9][Windows] M2M100ForConditionalGeneration and MBartForConditionalGeneration failed, eager_2nd_run_OOM

### ISSUE THREAD

**Issue by kaileiyx** [intel/torch-xpu-ops#2069]
### 🐛 Describe the bug

M2M100ForConditionalGeneration and MBartForConditionalGeneration failed, eager_2nd_run_OOM 

xpu  train M2M100ForConditionalGeneration     
eager_2nd_run_OOM
xpu  train MBartForConditionalGeneration
eager_2nd_run_OOM

Reproducer
python benchmarks/dynamo/huggingface.py --accuracy --fp32 -d xpu -n10 --training  --only M2M100ForConditionalGeneration --cold-start-latency --backend=inductor
python benchmarks/dynamo/huggingface.py --accuracy --fp32 -d xpu -n10 --training  --only MBartForConditionalGeneration--cold-start-latency --backend=inductor

python benchmarks/dynamo/huggingface.py --accuracy --amp --amp-dtype float16 -d xpu -n10 --training  --only M2M100ForConditionalGeneration --cold-start-latency --backend=inductor
python benchmarks/dynamo/huggingface.py --accuracy --amp --amp-dtype float16 -d xpu -n10 --training  --only MBartForConditionalGeneration--cold-start-latency --backend=inductor

python benchmarks/dynamo/huggingface.py --accuracy --amp --amp-dtype bfloat16 -d xpu -n10 --training  --only M2M100ForConditionalGeneration --cold-start-latency --backend=inductor
python benchmarks/dynamo/huggingface.py --accuracy --amp --amp-dtype bfloat16 -d xpu -n10 --training  --only MBartForConditionalGeneration--cold-start-latency --backend=inductor

### Versions

PyTorch Commit: 2.9.0+xpu / c31a8186c1a43e0cd24521e8fe18dc650959ca19
OS: Microsoft Windows 11 Pro
CPU: Intel(R) Core(TM) Ultra 7 258V
GPU and Driver: Intel(R) Arc(TM) 140V GPU (16GB) 32.0.101.6913
oneAPI: intel.oneapi.win.deep-learning-essentials.product=2025.0.1+20

**Comment by mengfei25** [comment]
Should we find out why it is failed in the 2nd run but passed in the 1st run?
